src/mbgl/renderer/renderer_impl.cpp

```diff
@@ -655,6 +655,52 @@ std::vector<Feature> Renderer::Impl::queryRenderedFeatures(const ScreenLineStrin
 
     return result;
 }
+    
+std::vector<Feature> Renderer::Impl::queryShapeAnnotations(const ScreenLineString& geometry) const {
+    std::vector<const RenderLayer*> layers;
+    RenderedQueryOptions options;
+    for (const auto& layerImpl : *layerImpls) {
+        if (std::mismatch(layerImpl->id.begin(), layerImpl->id.end(),
+                          AnnotationManager::ShapeLayerID.begin(), AnnotationManager::ShapeLayerID.end()).second == AnnotationManager::ShapeLayerID.end()) {
+            if (const RenderLayer* layer = getRenderLayer(layerImpl->id)) {
+                layers.emplace_back(layer);
+            }
+        }
+    }
```

Comment:
The rest of this method is identical to `Renderer::Impl::queryRenderedFeatures`. Can you refactor the two methods to share the querying?